test/test_serialization.py

```diff
@@ -4425,6 +4426,43 @@ def test_get_unsafe_globals_in_checkpoint(self):
             finally:
                 torch._weights_only_unpickler._get_allowed_globals = old_get_allowed_globals
 
+    @parametrize("should_import", [False, True])
+    def test_load_njt_weights_only(self, should_import):
+        with tempfile.NamedTemporaryFile() as f:
+            njt = torch.nested.nested_tensor([[1, 2, 3], [4, 5]], layout=torch.jagged)
+            torch.save(njt, f)
+            import_string = "import torch._dynamo" if should_import else ""
+            # torch.nested is actually imported by default, so only torch._dynamo needs to be imported
+            script = f"""
+import torch
+{import_string}
+x = torch.load("{f.name}", weights_only=True)
+"""
```

Comment:
It feels like you are repeating the same pattern here as in https://github.com/pytorch/pytorch/pull/140740, do you mind moving this logic into common_utils and call it something like `def _attempt_load_from_subprocess(file: pathlib.Path, exected_failure_message: str| None = None ) -> None`